src/builder.rs

```diff
@@ -231,8 +231,14 @@ impl NodeBuilder {
 	}
 
 	/// Sets the IP address and TCP port on which [`Node`] will listen for incoming network connections.
-	pub fn set_listening_address(&mut self, listening_address: SocketAddress) -> &mut Self {
-		self.config.listening_address = Some(listening_address);
+	pub fn set_listening_addresses(
+		&mut self, listening_addresses: Vec<SocketAddress>,
+	) -> &mut Self {
+		if listening_addresses.len() > 100 {
+			panic!("Listening addresses can not exceed 100");
```

Comment:
I chose 100, because the `peer_handler` has a limit for 100 addresses on [`broadcast_node_announcement`](https://github.com/lightningdevkit/rust-lightning/blob/d5a0eb4270eae847763b4b13439fdfe9a1302a99/lightning/src/ln/peer_handler.rs#L2429). Will update the function to return an error.